llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -116,31 +116,86 @@ void GCNSchedStrategy::initialize(ScheduleDAGMI *DAG) {
                     << ", SGPRExcessLimit = " << SGPRExcessLimit << "\n\n");
 }
 
+static bool canUsePressureDiffs(SUnit *SU) {
+  if (SU->isInstr()) {
+    // Cannot use pressure diffs for subregister defs or with physregs, it's
+    // imprecise in both cases.
+    for (const auto &Op : SU->getInstr()->operands()) {
+      if (!Op.isReg() || Op.isImplicit())
```

Comment:
Why skip implicit operands? (plus I never remember which sense of implicit operands MachineOperand::isImplicit returns true for)

Reply:
We often have virtual regs with implicit physreg operands, like implicit exec.
Implicit operands liveness, for some reason, cause no problems as far as I can see.
I ran the expensive check below on:

- A large full LTO module (120MB of bitcode)
- Internal CI
- Lit tests

And ignoring implicit operands works fine. If we didn't ignore them we'd probably end up using the expensive path much more often